### DISCUSSION TITLE
`if @version.nil? || @version.null?`

### DISCUSSION THREAD

**Discussion by abitrolly** [Homebrew/discussions#5135] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.8-126-g23c2eab
ORIGIN: https://github.com/Homebrew/brew
HEAD: 23c2eabf9a874a47e571f40a4f40daefdfad4b19
Last commit: 5 minutes ago
Core tap HEAD: f7d778e1bae8d8c42fdce3e3edb32c97d79f28ec
Core tap last commit: 8 hours ago
Core tap JSON: 16 Feb 05:47 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: quad-core 64-bit haswell
Clang: N/A
Git: 2.43.0 => /bin/git
Curl: 8.2.1 => /bin/curl
Kernel: Linux 6.7.4-200.fc39.x86_64 x86_64 GNU/Linux
OS: Fedora release 39 (Thirty Nine)
Host glibc: 2.38
/usr/bin/gcc: 13.2.1
/usr/bin/ruby: 3.2.2
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.                                                   /3,8s
```

### Description of issue

What `.null?` check does here? I can't find that Ruby has that syntax.

https://github.com/Homebrew/brew/blob/645cf35ae11105a71cf220fdaf0134774132c810/Library/Homebrew/formula_creator.rb#L71

**Comment by ZhongRuoyu** [comment, marked as answer]
See:

- https://rubydoc.brew.sh/Version.html#null%3F-instance_method
- https://github.com/Homebrew/brew/blob/a6d539cf11a0db5a2c0f6fb78ff1ed09bf576a93/Library/Homebrew/version.rb#L533-L536
  ```ruby
  sig { returns(T::Boolean) }
  def null?
    version.nil?
  end
  ```

**Reply by abitrolly** [reply]
Oh, that's confusing. At first I thought it is just an alias for `.nil?`, but `.nil?` means "no object", so when there is object, there `.null?` method means "version is empty".

Is there a way to implement "emptyness" check in Ruby to check `if @version` instead of `if @version.nil? || @version.null?` ?

**Reply by ZhongRuoyu** [reply]
> Is there a way to implement "emptyness" check in Ruby to check `if @version` instead of `if @version.nil? || @version.null?` ?

Maybe not, at least not in a straightforward way to my knowledge: https://stackoverflow.com/q/14444975
